lib/osf-components/addon/components/resources-list/component.ts

```diff
@@ -0,0 +1,46 @@
+import { A } from '@ember/array';
+import { assert } from '@ember/debug';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+
+import RegistrationModel from 'ember-osf-web/models/registration';
+import ResourceModel from 'ember-osf-web/models/resource';
+
+interface ResourcesListArgs {
+    registration: RegistrationModel;
+}
+
+export default class ResourcesListComponent extends Component<ResourcesListArgs> {
+    @tracked resourcesList: ResourceModel[] = A([]);
+    @tracked resourcesPage = 1;
+    @tracked totalResources = 0;
+
+    get hasMore() {
+        return this.totalResources > this.resourcesList.length;
+    }
+
+    get isLoading() {
+        return taskFor(this.loadResources).isRunning;
+    }
+
+    @task
+    @waitFor
+    async loadResources() {
+        const resourceList = await this.args.registration.queryHasMany('resources',
+            {
+                page: this.resourcesPage,
+            });
+        this.totalResources = resourceList.meta.total;
+        this.resourcesList.pushObjects(resourceList);
+        this.resourcesPage += 1;
+    }
+
+    constructor(owner: unknown, args: ResourcesListArgs) {
+        super(owner, args);
+        assert('ResourcesListComponent requires a registration', this.args.registration);
```

Comment:
I remember having some trouble with this pattern on the files page. I thiiink it was related to when a target was provided but was invalid, it didn't handle the case very well. It may be fine if other error handling will properly 404 or whatever, but it might be worth verifying if that's the case.